lib/ansible/modules/crypto/luks_device.py

```diff
@@ -744,15 +741,32 @@ def run_module():
     # luks add key
     if conditions.luks_add_key():
         if not module.check_mode:
+            # check if key already exists
+            key_exists = False
+            slotname = str(os.stat(conditions.device).st_mtime)
             try:
-                crypt.run_luks_add_key(conditions.device,
-                                       module.params['keyfile'],
-                                       module.params['passphrase'],
-                                       module.params['new_keyfile'],
-                                       module.params['new_passphrase'])
+                key_exists = crypt.run_luks_open(conditions.device,
+                                                 module.params['new_keyfile'],
+                                                 module.params['passphrase'],
+                                                 slotname)
+            except ValueError:
+                pass
+
+            try:
+                if key_exists:
+                    crypt.run_luks_close(slotname)
+                else:
+                    crypt.run_luks_add_key(conditions.device,
+                                           module.params['keyfile'],
+                                           module.params['passphrase'],
+                                           module.params['new_keyfile'],
+                                           module.params['new_passphrase'])
             except ValueError as e:
                 module.fail_json(msg="luks_device error: %s" % e)
-        result['changed'] = True
+
+            if key_exists is False:
+                result['changed'] = True
+                result['name'] = slotname
         if module.check_mode:
```

Comment:
This doesn't work anymore, since `result['changed']` is no longer set to `True`.